docs/initialization-steps-and-the-related-tracer.org

```diff
@@ -0,0 +1,624 @@
+
+    First it needs to select a starting ledger state with which it will
+    initialize ~LgrDB~. In order to do that, list of all available snapshots is
+    retrieved from the disk. Two things may happen:
+
+    + there are no valid snapshots available
+    + there is at least one valid snapshot available
+
+**** there are no valid snapshots available
+
+     If there are no valid snapshots available, it will:
+     + fire an event ~ReplayFromGenesis~
+     + create a ~LedgerDB~ anchored at Genesis
+     + start initialization with that ~LedgerDB~
+
+**** there is at least one valid snapshot available
+
+     If there is at least one valid snapshot available, it will:
+
+     + fire an event ~ReplayFromSnapshot~
+     + create a ~LedgerDB~ anchored at that snapshot
+     + start initialization with that ~LedgerDB~
+
+    Initialization builds a ~LedgerDB~ by folding over blocks returned by the
+    ~ImmutableDB~, starting from the block corresponding to the anchor of the
+    initial ~LedgerDB~ going up to the tip of the ~ImmutableDB~.
+    Each block is then validated, applied to existing ledger state, and extend
+    the current ~LedgerDB~ with a new state.
+    This process is called a replay.
+
+    For each block that we replay a ~ReplayedBlock~ is fired
+
+** initialize chain selection
+
+
+   Perform the initial chain selection based on the tip of the ImmutableDB and
+   the contents of the VolatileDB.
+   While executing, it will fire events described in [[*~TraceInitChainSelEvent~ - a.k.a init chain selection's events][~TraceInitChainSelEvent~ - a.k.a init chain selection's events]]
+
+   List of candidates is created based on blocks that live in the
+   ~VolatileDB~. If that is list empty, meaning that there are no candidates
+   available, an empty chain fragment (anchored at the ~ImmutableDB~ tip) is
+   returned. Otherwise a chain selection is initialized.
+
+   The chain selection algorithm will gather all chain candidates, sort them
+   (in order from best to worst) and then it will validated the first
+   candidate that is on the list. If candidate happens invalid it will move to
+   the next one, otherwise it will return it.
+   While the validation happens few (more then one during single validation)
+   events might be fired:
+   + ~ValidCandidate~ if the validation was successful
+   + ~InvalidBlock~
+   + ~CandidateContainsFutureBlocks~ - Candidate contains headers from the
+     future which do no exceed the clock skew.
+   + ~CandidateContainsFutureBlocksExceedingClockSkew~ - Candidate contains
+     headers from the future which exceed the clock skew, making them invalid.
+
+* Node execution logs
+
+  Reading the [[*Node execution logs][Node execution logs]] we can clearly see that there is a significant
+  time delay between ~TraceOpenEvent OpenedVolatileDB~ ("Opened vol db") and ~ReplayFromSnapshot~
+  ("Replaying ledger from snapshot DiskSnapshot").
+
+
+#+CAPTION: Node logs, with DEBUG mode disabled
+#+BEGIN_SRC
```

Comment:
- Would it make sense to describe what we should look at in the log below?
- What is the time delay we're looking at?  14:03:20.23 - 14:02:57.61? This is around 23 seconds? Is that the significant delay mentioned above?
- We also see a large delay between the last two chain extension messages. Does this indicate a problem as well?
- The does not show any of the events mentioned here. Would it make sense to run the node with a verbosity setup that shows these events?

Reply:
> * Would it make sense to describe what we should look at in the log below?

Can you propose something? I don't know what you mean by this.

> * What is the time delay we're looking at?  14:03:20.23 - 14:02:57.61? This is around 23 seconds? Is that the significant delay mentioned above?

It's mentioned above "  Reading the [[*Node execution logs][Node execution logs]] we can clearly see that there is a significant  time delay between ~TraceOpenEvent OpenedVolatileDB~ ("Opened vol db") and ~ReplayFromSnapshot~
  ("Replaying ledger from snapshot DiskSnapshot")."

> * We also see a large delay between the last two chain extension messages. Does this indicate a problem as well?

Not really. Chain was extended because new block was forged. We don't have control over it.

> * The does not show any of the events mentioned here. Would it make sense to run the node with a verbosity setup that shows these events?

Yes, it does. But lets make it under  a new PR so that this document gets into the repository.